From 6b899b33b9062bdc37cd85474fdc4106fc850924 Mon Sep 17 00:00:00 2001
From: Erez Shaul <eshaul@habana.ai>
Date: Tue, 08 Oct 2024 10:03:12 +0300
Subject: [PATCH] [SW-190033] boot_desc_init: check actual struct size

hl_fw_boot_ptrs now includes new fields of version & size.
With this information we can detect special cases in which the flash
content is more updated than the firmware that is running.
Is such cases, if we detect that the actual size is bigger, and we
are going to lose the struct tail we will warn, but continue to use
the fields that the firmware knows.

Change-Id: I9a5d6d07a84fdc1ebaa988d33fe84c206870da4f
Signed-off-by: Erez Shaul <eshaul@habana.ai>
---

diff --git a/lib/common/boot_desc.c b/lib/common/boot_desc.c
index 4c2f033..5b11c49 100644
--- a/lib/common/boot_desc.c
+++ b/lib/common/boot_desc.c
@@ -123,6 +123,10 @@
 
 	/* Read the main boot pointers structure */
 	spi_flash_copy(&boot_ptrs, spi_off, BOOT_PTRS_SZ, "BOOT PTRS");
+	if (boot_ptrs.version > 0) {
+		if (boot_ptrs.boot_ptrs_sz > BOOT_PTRS_SZ)
+			preboot_puts("Warning: boot_ptrs size is bigger than expected\n");
+	}
 	boot_desc_print_magic("FW boot PTRs", &boot_ptrs.magic);
 
 	if (boot_ptrs.magic.val != HL_FW_BOOT_MAGIC)
